File: iscp/lab1/prelab.java
import java.util.*;

class prelab
{
    public static void main(String args[])
    {
        Stack<Character> stack = new Stack<Character>();
        Scanner input = new Scanner(System.in);
        System.out.println("Please enter the no of commands");
        int number_of_commands = input.nextInt();
        for(int i=0;i < number_of_commands;i++)
        {
            String line = input.next();
            if(line.charAt(0) == '1')
            {
                stack.push(line.charAt(1));
            }
            else if(line.charAt(0) == '2')
            {
                System.out.println("The elemnent that is popped is "+stack.pop());
                
            }
            else if(line.charAt(0) == '3')
            {
                System.out.println("The largest element is "+max_element(stack));
            }
        }
        input.close();
        
    }
    static char max_element(Stack<Character> stack_g)
    {
        char max_number = 0;
        char ok = 0;
        Stack<Character> stack = (Stack<Character>) stack_g.clone();
        try
        {
            ok = stack.pop();
            while(ok != 0)
            {
                if(ok > max_number)
                {
                    max_number = ok;
                }
                ok = stack.pop();
            }
        }
        catch(Exception e)
        {
            return ok;
        }
        return ok;
    }

}

File: iscp/lab1/prelab.py
class Stack(list):
    def __init__(self):
        self.stack = []
    def push(self,data : int) -> None:
        self.stack.append(data)
    def pop(self):
        try:
            return self.stack.pop()
        except:
            return None
def find_largest_in_stack(stack : Stack):
    max_element = 0
    ok = stack.pop()
    while(ok is not None):
        if(ok >  max_element):
            max_element = ok
        ok = stack.pop()
    return max_element
def main():
    no_of_commands = input("Please enter the number of elements")
    command_list = input("Please enter the elements seperated by ' '")
    commands = [i for i in command_list.split()]
    stack = Stack()
    for command in commands:
        if(command[0] == "1"):
            stack.push(int(command[1]))
            print("Added element ",command[1])
        elif(command[0] == "2"):
            print("Popper ",stack.pop())
        elif(command[0] == "3"):
            print("the largest element is ",find_largest_in_stack(stack))
main()

